enctests/testframework/generatetests.py

```diff
@@ -0,0 +1,149 @@
+# This is for generating tests from the markdown code.
+import os
+import argparse
+import yaml
+import re
+
+try:
+    from yaml import CSafeLoader as SafeLoader
+    from yaml import CSafeDumper as SafeDumper
+
+except ImportError:
+    from yaml import SafeLoader, SafeDumper
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--root',
+        action='store',
+        default='..',
+        help='Where to look for *.md files to generate the config file.'
+    )
+
+    parser.add_argument(
+        '--output-config',
+        action='store',
+        dest='output_config_file',
+        default="test_configs/documentation_tests.yml",
+        help='Specify output file to write test wrappers to.'
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+def main():
+    args = parse_args()
+    rootdir = os.path.abspath(args.root)
```

Comment:
We've used "pathlib" for all path handling in other parts of the code base. Should we be consistent and use it here as well?

Reply:
Should be addressed.